src/base/memory_monitor_sr.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * memory_monitor_sr.cpp - implementation of memory monitoring manager
+ */
+
+#include "perf.hpp"
+
+#include "error_manager.h"
+
+#include "memory_monitor_sr.hpp"
+#if defined (SERVER_MODE)
+#include "log_impl.h"
+#endif
+
+#include <stdexcept>
+
+#include <cstring>
+
+#ifdef SERVER_MODE
+namespace cubperf
+{
+  MMM *MMM_global = NULL;
+#if defined(NDEBUG)
+  std::atomic<uint64_t> test_meminfo;
+#endif
+  MMM_module::MMM_module (char *name, MMM_comp_info *info)
```

Comment:
```suggestion
  MMM_module::MMM_module (const char *name, MMM_comp_info *info)
```

if the name is not changed, setting this type to const looks better

Reply:
I declared _const char *_ in .hpp file but it was not reflected here, so I modified it.